pkg/scale/scale.go

```diff
@@ -112,9 +115,19 @@ func (cc *cruiseControlScaler) Status(ctx context.Context) (CruiseControlStatus,
 		return CruiseControlStatus{}, err
 	}
 
+	result := resp.Result
+	// if the execution takes too much time, then cruise control converts the request into async
+	// and returns the taskID
+	if result == nil {
+		result, err = cc.waitForTaskCompletion(ctx, resp.TaskID)
```

Comment:
is this blocking the whole `Status` function until in completes?
asking cause it may take longer and we should not wait and instead return a `Pending` status and let the operator feedback loop to continue

Reply:
Note that if status is non-ready, then the reconciliation loop is interrupted and [requeued for later](https://github.com/adobe/koperator/blob/master/controllers/cruisecontroloperation_controller.go#L153).

Do you suggest adding a new [CruiseControlTaskOperation](https://github.com/adobe/koperator/blob/master/api/v1alpha1/common_types.go#L38) and then storing it as a k8s object, so at the next reconciliation we will check the status of async task?

Something like:
1. If CC status api is converted to async, then store the state in a [cruisecontroloperations](https://github.com/adobe/koperator/blob/master/config/base/crds/kafka.banzaicloud.io_cruisecontroloperations.yaml) object
2. return `Pending` so the reconciliation is requeued for later
3. On the next reconciliation execution, instead of calling the `Status` CC API again, we will call instead the `USER_TASKS` CC API endpoint to check if our async task finished. And return either `Pending` or the actual `CC` status

Reply:
Or just to return `Pending` and then at the next reconciliation to try again?

Reply:
Hey @amuraru, I created a draft PR where I use the approach with [cruisecontroloperations](https://github.com/adobe/koperator/blob/master/config/base/crds/kafka.banzaicloud.io_cruisecontroloperations.yaml)  described above.
Please take a look and let me know if you think it's a better solution and it makes sense for me to finish it.
Draft PR link: https://github.com/aguzovatii/koperator/pull/1